### PR TITLE
[MODORDERS-730] Item not appearing under holdings after instance connection change

### PR DESCRIPTION
**Pull request by Abdulkhakimov**
## Purpose
When in  PoLine receiving workflow set as Independent items is not updated.

## Approach

- Replacing logic of checking need to update item when using patch method with holdingsOperation - find or replace. We should just compare current and new holdings.

## Learning
[MODORDERS-730](https://issues.folio.org/browse/MODORDERS-730)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java [hunk 1]
```diff
@@ -85,7 +86,7 @@ private CompletableFuture<Void> findOrCreateHoldingsAndUpdateItems(OrderLineUpda
               .thenCompose(newHoldingId -> {
                 holder.addHoldingRefsToStoragePatchOrderLineRequest(holdingId, newHoldingId);
                 CompositePoLine compositePoLine = PoLineCommonUtil.convertToCompositePoLine(holder.getStoragePoLine());
-                if (PoLineCommonUtil.isItemsUpdateRequired(compositePoLine)) {
+                if (ObjectUtils.notEqual(holdingId, newHoldingId)) {
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:89]: PoLineCommonUtil.isItemsUpdateRequired(compositePoLine) - this method used in the current class twice, do we need to replace both usages?

Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:89]: Hi @SerhiiNosko. I can not give definitely answer to your question due my poor knowledge the domain. Better I'll wait @grabsefx and @AndreiMPV.

Comment (Abdulkhakimov) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:89]: @AndreiMPV I decide to remove any conditions for "create holdings" case. No make sense checking equality holding Ids when we have holdings operation with value "CREATE"

Comment (SerhiiNosko) [reply src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java:89]: It makes sense, for new holding  holdingId will always will be different, so this check not needed

**Review by AndreiMPV — APPROVED** [review]

**Pull request merged**